メモアプリの実装

junichiohta opened the pull request:
- Memoクラス: メモデータを管理
- MemoRepository: データの永続化を担当
- MemoApp: アプリケーションロジック
- CLIインターフェース実装

junichiohta committed:
メモアプリの実装

- Memoクラス: メモデータを管理
- MemoRepository: データの永続化を担当
- MemoApp: アプリケーションロジック
- CLIインターフェース実装

yuuu commented on 04.class/lib/MemoApp.js:
```diff
@@ -0,0 +1,96 @@
+      return;
+    }
+
+    const choices = memos.map((memo) => ({
```
この処理が複数箇所に存在しているので、メソッドに切り出してはどうでしょうか？

yuuu commented on 04.class/lib/MemoRepository.js:
```diff
@@ -0,0 +1,47 @@
+  }
+
+  save(memos) {
+    const data = JSON.stringify(memos, null, 2);
```
`2` という数字が何だったか一瞬出てこなかったです。
定数化しておくとわかりやすいですね 👍

yuuu commented on 04.class/package.json:
```diff
@@ -1,14 +1,23 @@
+  "type": "module",
+  "dependencies": {
+    "enquirer": "^2.4.1",
+    "minimist": "^1.2.8"
```
こちらは現状使っていないようなので、不要であれば削除しましょう

junichiohta committed:
レビュー指摘の修正

- 重複コードをcreateChoices()メソッドに切り出し
- JSON.stringify()のマジックナンバーを定数化
- 未使用のminimistを削除
